### ISSUE TITLE
Question on setting/options for use case

### ISSUE THREAD

**Issue by tyfoon** [tolwi/hassio-ecoflow-cloud#36]
Thanks for creating this integration.

Quite a lot of settings are exposed. I have a question about that;

I want to use my Delta 2 to continuously power my ‘IT environment’ which is basically router, few POE switches, few PI’s, Few Intel NUC. Overall they draw about 125w-140w continuous.

I want to charge the Delta 2 (triggered by Home Assistant) when prices are low and use the Delta 2 stored energy when prices are high + in case of power outage. In case the battery is empty it should just pass through the power from the mains.

I currently also have a UPS in between but will pull that obviously when above is up and running. 

Building the automations is no issue but what setting/controls can I best use on the Delta 2 to facilitate this?

**Comment by tyfoon** [comment]
Thanks Raphael, your setup sounds indeed identical. 

Few questions pop up;

- What happens if the automation for some reason (cloud down, etc) does not reach the EcoFlow when it is not pulling from the mains? Does it automatically fall back to mains when battery is drained?
- Do you have 'AC always on' enabled?

I'm also thinking to leave my UPS in between for now. This will also allow me to use the Ecoflow for something else  if I need it without my whole house going down. 

Thanks!

**Comment by raphaelsavina** [comment]
Just to be clear:

- D2 at 100%, priority solar OFF: current will only pass through, no drain on D2
- D2 20%-100%, priority solar ON: current will be drained from D2, to feed what is connected (125w-140w in your setup) 
- D2 20%-100%, priority solar OFF: current will be drained from main, enough to feed what is connected (125w-140w in your setup) + AC charging power (200-1200W that can be setup in your automation).
- D2 >20%: Priority solar will be ignored and the drain from power would be as above (125w-140w + 200-1200w)

There is a risk using a cloud integration, but in my setup, I have other integrations linked to cloud services (not my choice :-( ) so if any system is offline I will have issues.
 
Yes, AC is on most of the time, it is off only when I'm not using the Delta2 to keep the charge (the D2 is also used in my camper van on weekend away).

**Comment by tyfoon** [comment]
Thanks. I'm testing this now. But with prio solar charging on and charge at 98% (see screenshot and validated in app) it still seems to be passing through instead of draining from the D2.

<img width="1554" alt="Screenshot 2023-03-30 at 14 47 20" src="https://user-images.githubusercontent.com/22179507/228841160-38a67874-99e6-442f-8e76-0d024422af16.png">

**Comment by raphaelsavina** [comment]
I actually have 2 old solar panels (120W total) always connected so maybew there is something to do with the Ecoflow XT60 cable.

**Comment by tyfoon** [comment]
Mhh. That could be. Does it keep working also when the panels deliver 0 energy or does it jump to AC Pass through when the panels deliver 0?

**Comment by raphaelsavina** [comment]
It does not matter if there is or not solar production, at 0W it still does not take from AC.

Also, just to show you, with prio solor, AC in is at 0:

![Screenshot 2023-04-04 at 13 50 52](https://user-images.githubusercontent.com/540382/229783233-0e1f4c72-c61c-4609-b339-56dc35844690.png)

**Comment by raphaelsavina** [comment]
Yes, I have an "i" version. The difference is that the amp solar input can be higher can with the car (not "i") but it might be used as well by the D2 software to block AC. 
Extra info: https://www.youtube.com/watch?v=T1qFCzeCKF4

**Comment by tyfoon** [comment]
Well spotted. Cable ordered. Should have it in the next few days. Wondering if it will need a resistor or anything else connected to 'fake' that there is a panel connected at the ends of the cable.

**Comment by tyfoon** [comment]
Arghh. Official cable for solar charging (the correct one for the D2 with XT60i) + even connected panel (although unfortunately no sun right now) but still no luck. Pro solar on or off; It keeps using pass through.

Do you have any other switches/settings changed? Could you maybe post a screenshot?

p.s. I also mailed Ecoflow but no response yet

**Comment by raphaelsavina** [comment]
Hi @tyfoon I just took these 2 shots now.
What I have behind the D2 is taking just above 250W and charging is setup at 243W. 
Without prio solar, AC in is 521W, out 280W so ther is pass through
With prio solar: AC in is a 0W, out 245W, solar in 42W (not much sun right now)

![Screenshot 2023-04-06 at 15 56 58](https://user-images.githubusercontent.com/540382/230400826-7ec722df-ebb8-4a0a-936d-e45f57ff59bb.png)
![Screenshot 2023-04-06 at 15 57 39](https://user-images.githubusercontent.com/540382/230400833-ffc8b8ec-a866-481d-8127-119465831a3f.png)

I hope it helps.

Raphaël

**Comment by tyfoon** [comment]
Thanks for sharing this. really appreciated. I see indeed nothing that is different from my settings although you seem to have an extra battery but I cannot imagine that plays a role.

Even tough I have a panel connected, the sun today is 0 so also 0 Watt from panel.  Last thing I can try is to see if there is actual sun and current coming from the panel, it might switch something internally.

Does your solar prio switch continue it's function even if solar input is actually 0?

**Comment by raphaelsavina** [comment]
Yes, the fact that there is sun or not does not affect. Now, the panels are always connected, so maybe the XT60i has an impact.
